Name refactoring

brianhlee opened the pull request:
I left out a commit

Commit:
renamed aws resouces to follow naming conventions

didn't know what to name the gcp resource, so left them as is for now

Commit:
edit

hosted zone was fine as is
vpc -> vpc instance
compute -> compute engine
storage -> bucket

Commit:
Merge remote-tracking branch 'upstream/develop' into name-refactoring

Commit:
change vpc class name

Commit:
fixed changes that I missed in VPCInstance

flavor name
and one place i missed for changing the class name

bhl5kj commented:
realized I need to update docs for this. will do that real quick

bhl5kj commented:
actually, i would have to do it again when azure blob gets merged. I'll wait for one to be merged then update the other to have both changes

Commit:
Merge remote-tracking branch 'upstream/develop' into name-refactoring

Commit:
docs fixes

applied the name-refactoring to the docs

changed the subsubsections (particles) in the particle list to be different from the sub sections (aws, az, gcp)
